src/org/traccar/protocol/OwnTracksProtocolDecoder.java

```diff
@@ -46,12 +49,53 @@ protected Object decode(
         JsonObject root = Json.createReader(
                 new StringReader(request.content().toString(StandardCharsets.US_ASCII))).readObject();
 
-        if (!root.containsKey("_type") || !root.getString("_type").equals("location")) {
+        if (!root.containsKey("_type")) {
+            sendResponse(channel, HttpResponseStatus.OK);
+            return null;
+        }
+        if (!root.getString("_type").equals("location")
+            && !root.getString("_type").equals("lwt")) {
+            sendResponse(channel, HttpResponseStatus.BAD_REQUEST);
+            return null;
+        }
+
+        Position position = new Position();
+        String uniqueId;
+
+        if (root.containsKey("topic")) {
+            uniqueId = root.getString("topic");
+            if (root.containsKey("tid")) {
+                position.set("tid", root.getString("tid"));
+            }
+        } else {
+            uniqueId = root.getString("tid");
+        }
+        DeviceSession deviceSession = getDeviceSession(channel, remoteAddress, uniqueId);
+        if (deviceSession == null) {
+            sendResponse(channel, HttpResponseStatus.BAD_REQUEST);
+            return null;
+        }
+
+        if (root.getString("_type").equals("lwt")) {
+            Context.getConnectionManager().updateDevice(deviceSession.getDeviceId(), Device.STATUS_OFFLINE, new Date());
```

Comment:
You should never do this in decoder.

Reply:
What would be the right way to indicate a Device gone off-line when using HTTP?

Reply:
I understand your view.

Still I am a bit sad because the offline information is available in the originating system and the code above did reflect this clearly in traccar.

Maybe sometime in the future we will have an MQTT Protocol class which will allow to manage device state...